packages/itwinui-react/src/core/Popover/Popover.tsx

```diff
@@ -154,15 +202,32 @@ export const usePopover = (options: PopoverOptions & PopoverInternalProps) => {
   });
 
   const interactions = useInteractions([
-    useClick(floating.context, { enabled: !!trigger.click }),
-    useDismiss(floating.context, { outsidePress: closeOnOutsideClick }),
+    useClick(floating.context, {
+      enabled: !!mergedInteractions.click,
+      ...(mergedInteractions.click as object),
+    }),
+    useDismiss(floating.context, {
+      enabled: !!mergedInteractions.dismiss,
+      outsidePress: closeOnOutsideClick,
+      bubbles: tree != null, // Only bubble when inside a FloatingTree
+      ...(mergedInteractions.dismiss as object),
+    }),
     useHover(floating.context, {
-      enabled: !!trigger.hover,
+      enabled: !!mergedInteractions.hover,
       delay: 100,
-      handleClose: safePolygon({ buffer: 1 }),
+      handleClose: safePolygon({ buffer: 1, requireIntent: false }),
```

Comment:
While not directly relevant for this PR, I noticed safe polygon only works for the very last item in a menu. I investigated further and found that it starts working correctly when [`position: relative`](https://github.com/iTwin/iTwinUI/blob/e2e6ba7abbb46ac99ed225775b56942b88d8da04/packages/itwinui-css/src/menu/mixins.scss#L46) is removed. 


Further investigation needed. (I believe the `position: relative` is necessary for [`ListItem.Action`](https://itwinui.bentley.com/docs/list#with-listitemaction) to work as intended).